sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -180,4 +184,60 @@ export class ApplicationController extends BaseController {
       );
     }
   }
+
+  /**
+   * Generic method to update all Student Application status from frontend.
+   * @param applicationId application id to be updated.
+   * @body payload contains the status, that need to be updated
+   */
+  @AllowAuthorizedParty(AuthorizedParties.student)
+  @Patch(":applicationId/update-status")
+  async updateStudentApplicationStatus(
+    @UserToken() userToken: IUserToken,
+    @Param("applicationId") applicationId: number,
+    @Body() payload: ApplicationStatusToBeUpdatedDto,
+  ): Promise<void> {
+    const student = await this.studentService.getStudentByUserId(
+      userToken.userId,
+    );
+    if (!student) {
+      throw new UnprocessableEntityException(
+        "The user is not associated with a student.",
+      );
+    }
+    const studentApplication =
+      await this.applicationService.getStudentApplicationStatus(
+        applicationId,
+        student,
+      );
+
+    if (!studentApplication) {
```

Comment:
I would recommend doing the student/application validation as below, in the same way already in place on method _getByApplicationId_.
```
 const application =
      await this.applicationService.getApplicationByIdAndUserName(
        applicationId,
        userToken.userName,
      );
    if (!application) {
      throw new NotFoundException(
        `Application id ${applicationId} was not found.`,
      );
    }
```